### PR TITLE
Add a install script

### PR DESCRIPTION
**Pull request by Z33DD**
Works in Linux, Mac OS and FreeBSD

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -14,6 +14,21 @@ since only changes are stored.
...
+``$ chmod +x scripts/install``
+
+And after:
+
+``$./scripts/install``
+
+
+Probably ask you to enter your password, and it's done!
+
+
```
Comment (ThomasWaldmann) [inline README.rst]: no install docs in the readme please, move this to an appropriate section of the documentation.

File: scripts/Install [hunk 2]
```diff
@@ -0,0 +1,196 @@
+#!/bin/bash
+
+# Program information
+prog_name="BorgBackup"
+version="1.1.10"
```
Comment (ThomasWaldmann) [inline scripts/Install]: version number should be a commandline parameter.

nobody wants to update hardcoded version numbers each time a new release is made.

File: scripts/Install [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+signature="BSD License (BSD)"
```
Comment (ThomasWaldmann) [inline scripts/Install]: remove this, it is not needed here.

File: scripts/Install [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+git_url="https://github.com/borgbackup/borg"
+installation_path=/usr/local/bin
+
+# System information
+SYS=$(uname -s | awk '{print tolower($0)}')
+OS=$(awk '/DISTRIB_ID=/' /etc/*-release | sed 's/DISTRIB_ID=//' | tr '[:upper:]' '[:lower:]')
+ARCH=$(uname -m | sed 's/x86_//;s/i[3-6]86/32/')
+VERSION=$(awk '/DISTRIB_RELEASE=/' /etc/*-release | sed 's/DISTRIB_RELEASE=//' | sed 's/[.]0/./')
```
Comment (ThomasWaldmann) [inline scripts/Install]: I fear that this likely will break on some systems.

Did you try this on some linux systems, freebsd and mac osx?

File: scripts/Install [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    echo \n "We can't install the dependencies on this system"
+    echo "Try manually: wget gpg" \n
+
+  fi
+}
+
+upgrade_os(){
+  if [[ $OS == 'debian' || $OS == 'ubuntu' ]]; then
+    sudo apt-get update  > /dev/null
+    sudo apt-get upgrade -y  > /dev/null
```
Comment (ThomasWaldmann) [inline scripts/Install]: wtf?

your script started as a simple borg binary downloader some time ago and now it also upgrades the system it runs on.

this is getting way out of scope.

File: scripts/Install [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+# System information
+SYS=$(uname -s | awk '{print tolower($0)}')
+OS=$(awk '/DISTRIB_ID=/' /etc/*-release | sed 's/DISTRIB_ID=//' | tr '[:upper:]' '[:lower:]')
+ARCH=$(uname -m | sed 's/x86_//;s/i[3-6]86/32/')
+VERSION=$(awk '/DISTRIB_RELEASE=/' /etc/*-release | sed 's/DISTRIB_RELEASE=//' | sed 's/[.]0/./')
+
+
+
+install_all_dependencies(){
```
Comment (ThomasWaldmann) [inline scripts/Install]: these are dependencies required to build borg from source, please name the function so it indicates that.

File: scripts/Install [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+    libacl1-dev libacl1 \
+    build-essential libfuse-dev \
+    fuse pkg-config
+
+  elif [[ $OS == 'fedora' ]]; then
+    sudo dnf install python3 python3-devel python3-pip python3-virtualenv
+    sudo dnf install openssl-devel openssl
+    sudo dnf install libacl-devel libacl
+    sudo dnf install gcc gcc-c++
+    sudo dnf install redhat-rpm-config
```
Comment (ThomasWaldmann) [inline scripts/Install]: is this needed? why?

File: scripts/Install [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+
+  elif [[ $OSTYPE == 'darwin' ]]; then
+    brew install python3 openssl
+    brew install pkg-config
+    pip3 install virtualenv
+    brew cask install osxfuse
+
+  elif [[ "$OSTYPE" == "freebsd"* ]]; then
+    pkg install -y python3 openssl fusefs-libs pkgconf
+    pkg install -y git
```
Comment (ThomasWaldmann) [inline scripts/Install]: you install git here, but not on the other systems.

File: scripts/Install [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+    python3.4 -m ensurepip # to install pip for Python3
```
Comment (ThomasWaldmann) [inline scripts/Install]: this will only work on freebsd versions that have python 3.4 as default.

File: scripts/Install [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+    To use the mount command:
```
Comment (ThomasWaldmann) [inline scripts/Install]: did you forget to make this a comment?

File: scripts/Install [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+    echo 'fuse_load="YES"' >> /boot/loader.conf
+    echo 'vfs.usermount=1' >> /etc/sysctl.conf
+    kldload fuse
+    sysctl vfs.usermount=1
+
+  else; then
+    echo \n "We can't install the dependencies on this system"
+    echo "Try manually" \n
```
Comment (ThomasWaldmann) [inline scripts/Install]: end your sentences with a dot.

File: scripts/Install [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+
+  fi
+}
+
+install_dependencies(){
```
Comment (ThomasWaldmann) [inline scripts/Install]: this is the stuff you use to download/verify the borg binary, name your function appropriately.

File: scripts/Install [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+    sudo yum upgrade -y > /dev/null
+
+  elif [[ "$OSTYPE" == "freebsd"* ]]; then
+    freebsd-update fetch
+    freebsd-update install
+  fi
+}
+
+from_source(){
+  upgrade_os
```
Comment (ThomasWaldmann) [inline scripts/Install]: nope.

File: scripts/Install [hunk 14]
```diff
@@ -0,0 +1,196 @@
...
+  # install borg + dependencies into virtualenv
+  cd borg
+  pip install -r requirements.d/development.txt
+  pip install -r requirements.d/docs.txt  # optional, to build the docs
+  pip install -r requirements.d/fuse.txt  # optional, for FUSE support
+  pip install -e .  # in-place editable mode
+
+  # optional: run all the tests, on all supported Python versions
+  # requires fakeroot, available through your package manager
+  fakeroot -u tox
```
Comment (ThomasWaldmann) [inline scripts/Install]: nope.

this is to run the tests. we do this while developing borg and for release testing, but it is not intended to be part of a install-borg script.

File: scripts/Install [hunk 15]
```diff
@@ -0,0 +1,196 @@
...
+
+download_and_check_PGP(){
+  echo "Download... "
+  wget -q --show-progress $git_url/releases/download/$version/borg-$SYS$ARCH
+
+  echo "Download GPG signature..."
+  wget -q --show-progress $git_url/releases/download/$version/borg-$SYS$ARCH.asc
+
+  gpg --keyserver hkp://keyserver.ubuntu.com:80
+  \ --recv-keys "6D5B EF9A DD20 7580 5747 B70F 9F88 FB52 FAF7 B393"
```
Comment (ThomasWaldmann) [inline scripts/Install]: kudos for putting the long/full key id here!

File: scripts/Install [hunk 16]
```diff
@@ -0,0 +1,196 @@
...
+
+  gpg --verify borg-$SYS$ARCH.asc
+
+
+  if [ $? -eq 0 ]
+  then
+    echo "All is well."
+  else
+    echo "Problem with signature."
```
Comment (ThomasWaldmann) [inline scripts/Install]: "Signature verification failed. Do not use this borg binary download!"

File: scripts/Install [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+    exit -1
+  fi
+}
+
+
+
+if [[ $1 = "--release" ]]; then
+  if [[ $2 = "" ]]; then
+    echo "[ Oops ] You missend the number release"
```
Comment (ThomasWaldmann) [inline scripts/Install]: typos...

File: scripts/Install [hunk 18]
```diff
@@ -0,0 +1,196 @@
...
+    exit -1
+  else
+    $version = $2
+  fi
+else
+  $version = "1.1.10"
```
Comment (ThomasWaldmann) [inline scripts/Install]: nope, not another place with a hardcoded version number.

File: scripts/Install [hunk 19]
```diff
@@ -0,0 +1,196 @@
...
+sudo chmod 755 $installation_path/borg
+
+read -p "Do you want to delete the installation files? [y/N]" delete_install_files
+
+if [[ $delete_install_files = 'Y' || $delete_install_files = 'y' | $delete_install_files = 'Yes' | $delete_install_files = 'yes' ]]; then
+  echo Delete borg-$SYS$ARCH
+  rm -f borg-$SYS$ARCH
+
+  echo Delete borg-$SYS$ARCH.asc
+  rm -f borg-$SYS$ARCH.asc
```
Comment (ThomasWaldmann) [inline scripts/Install]: you can do all this in one `rm` command and you do not need the `echo`.

**Comment by ThomasWaldmann** [comment]
@Z33DD did you see my feedback?

**Comment by Z33DD** [comment]
@ThomasWaldmann 

I read it now
Unfortunately, my time is running out 
But I'm happy to work on that little contribution whenever I can.
Now, honestly speaking, I have 0 experience developing for MacOS and BSDs.
I can do my best and write the script for Linux and you (or someone else) solve the problem of the other OS.

It's OK for you?

~Z33DD

**Comment by ThomasWaldmann** [comment]
Well, maybe read it again and comment on my feedback / suggestions?

**Comment by ThomasWaldmann** [comment]
Looks like this is going nowhere.

**Pull request closed without merging**